packages/flutter/lib/src/widgets/layout_builder.dart

```diff
@@ -223,6 +225,89 @@ class _RenderLayoutBuilder extends RenderBox with RenderObjectWithChildMixin<Ren
     if (child != null)
       context.paintChild(child, offset);
   }
+
+  bool _debugThrowIfNotCheckingIntrinsics() {
+    assert(() {
+      if (!RenderObject.debugCheckingIntrinsics) {
+        throw FlutterError(
+          'LayoutBuilder does not support returning intrinsic dimensions.\n'
+          'Calculating the intrinsic dimensions would require running the layout '
+          'callback speculatively, which might mutate the live render object tree.'
+        );
+      }
+      return true;
+    }());
+
+    return true;
+  }
+}
+
+/// Builds a sliver widget tree that can depend on its own [SliverConstraints].
+///
+/// Similar to the [LayoutBuilder] widget except its builder should return a sliver
+/// widget, and [SliverLayoutBuilder] is itself a sliver. The framework calls the
+/// [builder] function at layout time and provides the current [SliverConstraints].
+/// The [SliverLayoutBuilder]'s final [SliverGeometry] will match the [SliverGeometry]
+/// of its child.
+///
+///
+/// See also:
+///
+///  * [LayoutBuilder], the non-sliver version of this widget.
+class SliverLayoutBuilder extends _GenericLayoutBuilder<SliverConstraints> {
+  /// Creates a sliver widget that defers its building until layout.
+  ///
+  /// The [builder] argument must not be null.
+  const SliverLayoutBuilder({
+    Key key,
+    SliverLayoutWidgetBuilder builder,
+  }) : super(key: key, builder: builder);
+
+  /// Called at layout time to construct the widget tree. The builder must return
+  /// a non-null sliver widget.
+  @override
+  SliverLayoutWidgetBuilder get builder;
+
+  @override
+  _RenderSliverLayoutBuilder createRenderObject(BuildContext context) => _RenderSliverLayoutBuilder();
+}
+
+class _RenderSliverLayoutBuilder extends RenderSliver with RenderObjectWithChildMixin<RenderSliver>, _GenericRenderLayoutBuilder<SliverConstraints, RenderSliver> {
+  @override
+  double childMainAxisPosition(RenderObject child) {
+    assert(child != null);
+    assert(child == this.child);
+    return 0;
+  }
+
+  @override
+  void performLayout() {
+    layoutAndBuildChild();
+    child?.layout(constraints, parentUsesSize: true);
+    geometry = child?.geometry ?? SliverGeometry.zero;
+  }
+
+  @override
+  void applyPaintTransform(RenderObject child, Matrix4 transform) {
+    assert(child != null);
+    assert(child == this.child);
+    // No-op because child's offset is always (0, 0),
```

Comment:
We don't use the word 'no-op' in our documentation. maybe 'It does not require transformation because..."